# Phylogeographic-analysis-of-nine-spined-sticklebacks
Scripts and code used in manuscript "Detailed phylogeographic analysis of sticklebacks reveals challenges in genetic studies of admixed populations".

## Content

### Pipeline and associated scripts for each of the analyses in the manuscript.
1) [Mapping of raw reads and variant calling](https://github.com/XueyunF/nsp_phylogeo/blob/main/Variant_calling/variant%20calling%20pipeline.md) 
2) [Population structure and phylogentic analyses](https://github.com/XueyunF/nsp_phylogeo/blob/main/strucutre_phylo/Structure_phylogenetic_pipeline.md)
3) [Symmetry statistics and admixture tests](https://github.com/XueyunF/nsp_phylogeo/blob/main/Symmetry_Admixture/Analysis%20with%20ADMIXTOOLS%2C%20Dsuite%20and%20RASS.md)
4) Estimation of the divergence times with [MSMC2](https://github.com/XueyunF/nsp_phylogeo/blob/main/Est_Divergence_time/MSMC2%20pipeline.md), [moments](https://github.com/XueyunF/nsp_phylogeo/blob/main/Est_Divergence_time/Moments%20pipeline.md) and data simulation with [msprime](https://github.com/XueyunF/nsp_phylogeo/blob/main/simulations/msprime_script.md)
5) [Sex chromosome related analysis](https://github.com/XueyunF/nsp_phylogeo/blob/main/Sex_chr/sex%20chromosome%20sequecing%20coverage.md)

### Contigs included in nine-spine stickleback reference v6b can be found [here](https://github.com/XueyunF/nsp_phylogeo/blob/main/Variant_calling/NSP_V6b.fasta.fai)

### Detailed results of D-statistics, qpWave and f4-ratio tests can be found [here](https://github.com/XueyunF/nsp_phylogeo/tree/main/Symmetry_Admixture)
